server/macros/reuters_derive_dateline.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from superdesk.locators.locators import find_cities
+from bs4 import BeautifulSoup
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def reuters_derive_dateline(item, **kwargs):
+    """
+    It seems that most locations injected into the item by the parser are Bangalor
+    This function looks for a dateline in the article body an uses that.
+    :param items:
+    :return:
+    """
+    try:
+        html = item.get('body_html')
+        if html:
+            soup = BeautifulSoup(html)
+            pars = soup.findAll('p')
+            if len(pars) >= 2:
+                if 'byline' in item and item.get('byline') in pars[0].get_text():
+                    first = pars[1].get_text()
+                else:
+                    first = pars[0].get_text()
+                city, source, the_rest = first.partition(' (Reuters) - ')
+                if source:
+                    # sometimes the city is followed by a comma and either a date or a state
+                    city = city.split(',')[0]
+                    if any(char.isdigit() for char in city):
+                        return
+                    cities = find_cities()
```

Comment:
These 2-3 lines are copied across modules. Can we extract it to a function in ingest module?